__tests__/main.test.ts

```diff
@@ -1,24 +1,104 @@
-import {wait} from '../src/wait'
+jest.mock('@actions/exec')
+
+import {getDeployProps} from '../src/deploy-props'
+import {s3Update} from '../src/s3-update'
 import * as process from 'process'
 import * as cp from 'child_process'
 import * as path from 'path'
+import * as exec from "@actions/exec";
+
+
+test('getDeployProps version release', () => {
+  expect(getDeployProps("refs/tags/v1.2.3"))
+    .toEqual({deployPath: "version/v1.2.3", version: "v1.2.3"});
+})
+
+test('getDeployProps PT prefix branch', () => {
+  expect(getDeployProps("refs/heads/123456789-test-branch"))
+    .toEqual({deployPath: "branch/test-branch", branch: "test-branch"});
+})
 
-test('throws invalid number', async () => {
-  const input = parseInt('foo', 10)
-  await expect(wait(input)).rejects.toThrow('milliseconds not a number')
+test('getDeployProps PT suffix branch', () => {
+  expect(getDeployProps("refs/heads/test-branch-123456789"))
```

Comment:
FWIW: I usually use PT prefix branch names.

Reply:
As do I, but I think the matching code is designed to accept prefix or suffix.

Reply:
Yes, the test just above this one verifies that prefix IDs work too. I don't think I'd seen a suffix ID, but the s3_deploy.sh code this is replacing supported it.

Reply:
On a tangentially related note, I think that the matching code should be expanded to support the `#` at the beginning of the id -- asking for a friend who might have forgotten to remove the `#` after pasting from PT into the branch name and then been confused when the branch-matching code didn't remove it as expected. 😳

Reply:
I committed your updates and added tests for it.